xmodule/graders.py

```diff
@@ -501,3 +501,25 @@ def correctness_available(cls, show_correctness='', due_date=None, has_staff_acc
 
         # else: show_correctness == cls.ALWAYS
         return True
```

Comment:
@steff456 Sorry, I can't select unchanged lines. I think since the PR also implements the service for correctness, `ShowCorrectness.correctness_available` method of the class can be removed, as it duplicates the login in the new service. Make sure that all the places where this method is being called is replaced with the call to the method of the new service.

Reply:
I have a question here because I'm unsure on how to fix it. 

If I remove it and leave only the `ShowCorrectnessService` import, I end up with a circular import,

```
File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/subsection_grade.py", line 16, in <module>
    from xmodule.services import ShowCorrectnessService
  File "/edx/app/edxapp/edx-platform/xmodule/services.py", line 29, in <module>
    from lms.djangoapps.grades.api import signals as grades_signals
ImportError: cannot import name 'signals' from partially initialized module 'lms.djangoapps.grades.api' (most likely due to a circular import) (/edx/app/edxapp/edx-platform/lms/djangoapps/grades/api.py)
```

I'm not aware if I can call the `self.runtime.service` in the class `SubsectionGradeBase` defined at `subsection_grade.py` to avoid the circular import or if I should change the file where the `ShowCorrectnessService` is defined.

Reply:
@steff456

First, I see that in [`lms.djangoapps.grades.api`](https://github.com/openedx/edx-platform/blob/5a193348b7f5c053b0724284cae179e246558f66/lms/djangoapps/grades/api.py#L22) `signals` is imported from `lms.djangoapps.grades.signals`. So you can try replacing `from lms.djangoapps.grades.api import signals as grades_signals` with `from lms.djangoapps.grades.signals import signals`. This should help resolve the cyclical import issue. 

If not, the `grades_signals` is used only once in the whole `xmodule/services.py` file. You can move the import line into the method where it's used, because afaik it won't be evaluated during import, only when called, so should also resolve the issue, but it's a less desirable solution. If you do end up having to use it, leave a comment saying that the import is within the method to avoid cyclical import error.